Tfm 1.7.0 and MbedTLS 3.2.1 update

joerchan opened the pull request:
This PR adds support for TF-M 1.7.0 and MbedTLS 3.2.1, which currently have dependencies on each other and must be updated together. This should be changed for future releeases to decouple the two projects where possible.

joerchan committed:
tfm: Remove library model support

In TF-M 1.7.0 release the Library model has been removed.
Remove the library model support from zephyr before updating TF-M
version.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>
Signed-off-by: David Brown <david.brown@linaro.org>

joerchan committed:
samples: Add SFN model configurations

Add SFN model configurations to samples.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>
Signed-off-by: David Brown <david.brown@linaro.org>

ceolin commented:
I've tested it in my environment and looking CI logs, it seems that all errors are TF-M related.

microbuilder commented:
> I've tested it in my environment and looking CI logs, it seems that all errors are TF-M related.

Working on those now (Joakim, David and myself).

microbuilder committed:
samples: tfm_integration: Remove psa_firmware

Removes the `psa_firmware` sample, which is based on an older version
(0.7) of the FWU service from TF-M 1.6.0. This sample needs to be
refactored to use FWU 1.0, included in TF-M 1.7.0 and future releases.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>
Signed-off-by: David Brown <david.brown@linaro.org>

microbuilder committed:
samples: tfm_integration: Remove psa_crypto

Removes the `psa_crypto` sample from the current release, due to
PSA API conflicts that can not be immediately resolved between Zephyr's
instance of MbedTLS in the NS environment, and the TF-M PSA APIs
included when building with TF-M support.

PSA API changes upstream in MbedTLS 3.2.1 (used by TF-M 1.7.0), and
MbedTLS 3.1 (used in TF-M 1.6.0) need to be be resolved in both
upstream projects before this sample can be reintroduced.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>
Signed-off-by: David Brown <david.brown@linaro.org>

microbuilder committed:
manifest: Update to TF-M 1.7.0 and MBedTLS 3.2.1

Update TF-M from 1.6.0 to 1.7.0
Update MBedTLS from 3.1.0 to 3.2.1.

Updates the cmake wrapper for changes introduced in TF-M 1.7.0.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>
Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>
Signed-off-by: David Brown <david.brown@linaro.org>

microbuilder committed:
samples: tfm_integration: Update tfm_ipc for 1.7.0

Update the sample to be compatible with API changes introduced in
TF-M 1.7.0, adding a new direct call to the PSA Crypto API to generate
random data, and cleaning up existing functions for compatibility
sake.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>
Signed-off-by: David Brown <david.brown@linaro.org>

joerchan committed:
modules: trusted-firmware-m: Add TF-M connection based NS API source

Add TF-M connection based NCS API source file to build.
This file is needed when a secure service is using connection
based method.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>
Signed-off-by: David Brown <david.brown@linaro.org>

joerchan committed:
samples: tfm_secure_partition: Update sample for TF-M 1.7.0

Update TF-M secure partition sample for TF-M 1.7.0.
Removes the support for Library model in the sample.
Updates to using PSA framework 1.1.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>
Signed-off-by: David Brown <david.brown@linaro.org>

microbuilder committed:
samples: tfm_ipc: Remove mps3_an547_ns support

The AN547 no longer functions with this TF-M sample, and has been
broken since TF-M 1.6.0 without CI catching the issue, since this sample
wasn't modified to cause a CI run on the affected target.

Removing this board from the sample until the board support can be
reworked.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>
Signed-off-by: David Brown <david.brown@linaro.org>

rerickson1 commented on west.yml:
```diff
@@ -241,15 +241,20 @@ manifest:
         - tee
+    - name: qcbor
+      revision: b0e7033268e88c9f27146fa9a1415ef4c19ebaff
+      path: modules/tee/tf-m/qcbor
```
Looks good. Just remove this as discussed in TSC meeting today.

joerchan replied:
Removed commit. Thank you.

microbuilder commented:
@rerickson1 We'll need one more force push once the MbedTLS PR is merged in, but I'll ping you for a review once that's done.

microbuilder commented:
@rerickson1 @d3zd3z @ceolin Have time for a review now that the PR is in it's final state?

microbuilder commented:
@SebastianBoe Have time for a review to try to get this in before feature freeze today?

SebastianBoe commented:
@microbuilder  Unfortunately not today.

microbuilder reviewed (approved):
Approving since I've tested locally, but not sure it should be considered as the missing +1 since I'm also a contributor on this PR.
